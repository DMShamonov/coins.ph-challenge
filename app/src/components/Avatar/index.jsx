// @flow

import * as React from 'react';
import classnames from 'classnames';

import Icon from 'components/Icon';

import type { PropsType } from './types.js.flow';

export default function Avatar({
  avatar = '',
  className,
}: PropsType): React.Node {
  return (
    <div className={classnames('avatar', className)}>
      {
        avatar
          ? <img src={avatar} width={48} height={48} alt="avatar" className="avatar__image" />
          : <Icon theme="avatar-placeholder" width={48} />
      }
    </div>
  );
}
