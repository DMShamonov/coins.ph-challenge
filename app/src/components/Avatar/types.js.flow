// @flow

export type PropsType = {|
  avatar?: string,
  className?: string,
|};
