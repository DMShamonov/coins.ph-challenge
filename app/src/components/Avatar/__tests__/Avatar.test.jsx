import React from 'react';
import { shallow } from 'enzyme';

import Avatar from '../index';

describe('`Avatar` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<Avatar />)).toMatchSnapshot();
  });

  test('with `avatar` prop', () => {
    expect(shallow(<Avatar avatar="__PATH_TO_AVATAR__" />)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<Avatar className="__CLASSNAME__" />)).toMatchSnapshot();
  });
});
