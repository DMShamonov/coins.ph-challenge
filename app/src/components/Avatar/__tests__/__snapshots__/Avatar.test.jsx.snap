// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`Avatar\` component should render correctly with \`avatar\` prop 1`] = `
<div
  className="avatar"
>
  <img
    alt="avatar"
    className="avatar__image"
    height={48}
    src="__PATH_TO_AVATAR__"
    width={48}
  />
</div>
`;

exports[`\`Avatar\` component should render correctly with \`className\` prop 1`] = `
<div
  className="avatar __CLASSNAME__"
>
  <Icon
    theme="avatar-placeholder"
    width={48}
  />
</div>
`;

exports[`\`Avatar\` component should render correctly with default props 1`] = `
<div
  className="avatar"
>
  <Icon
    theme="avatar-placeholder"
    width={48}
  />
</div>
`;
